### ISSUE TITLE
Debug View change page structure

### ISSUE THREAD

**Issue by puschie286** [bcit-ci/CodeIgniter4#761]
Hey, after pulling the new debug-view i have big problems with my site - the whole structure is changed

even the simple login page is changed in layout by the new debug-view ( in both development and production build )

Normal :
![normalstruct](https://user-images.githubusercontent.com/2429036/31327794-c0c0ee32-acd1-11e7-8acd-72745224614f.jpg)

With Debug-View :
![developmentstruct](https://user-images.githubusercontent.com/2429036/31327808-d2c5a672-acd1-11e7-8092-371c2dd8c53f.jpg)

As you can see, the content of the center-container div and header tag is now outside

**Comment by deathart** [comment]
![err_ci_dv_bug](https://user-images.githubusercontent.com/436468/31329958-aa09df5e-acdc-11e7-86a2-57c5fb013901.PNG)

**Comment by puschie286** [comment]
i guess there could be another problem - the additional divs can potentially break css child combinator. 

a possentioly solution would be to add the debug-view divs dynamicly on click on the view button in the debug bar ( but would still break js selections on activ debug-view )
( i dont know if this is the right place for this, maybe move to pullrequest [#759](https://github.com/bcit-ci/CodeIgniter4/pull/759/commits/fe3e8e4264671e8481817bf2721c762f3dac5912)

**Comment by natanfelles** [comment]
I have tested with custom css and Bootstrap and have not problems.

The divs [do not change your custom stylesheet](https://github.com/bcit-ci/CodeIgniter4/blob/6eedefb52e7dae5d14cd57dc0b156c73b5c446c8/system/Debug/Toolbar/Views/toolbar.css#L289).

But, ok, selectors with `parents` or `childs` can break the the styles or some javascript can not work.

How to "add the debug-view divs dynamicly on click on the view button in the debug bar"? It will be great but how to detect where a view starts and ends?

Uhmm... We can add html comments surrounding the views likes `<!-- start_view: /viewpath -->`
and `<!-- end_view: /viewpath -->` then add divs dynamically on this with javascript action on toolbar button click. But not sure if this will works with `parents` or `childs` if the page not be reloaded...

Please, show me the html source of your page and not the content of the browser inspector. It changes the code. "APPPATH"? I do not understand well this pictures, something is strange.

**Comment by natanfelles** [comment]
![captura de tela de 2017-10-09 19-21-10](https://user-images.githubusercontent.com/3011423/31360980-54f08568-ad27-11e7-86e6-b2c32fdc8ad0.png)

![captura de tela de 2017-10-09 19-21-15](https://user-images.githubusercontent.com/3011423/31360979-54ef676e-ad27-11e7-815b-095786b01137.png)

![captura de tela de 2017-10-09 19-21-29](https://user-images.githubusercontent.com/3011423/31360981-54f2b464-ad27-11e7-8c15-99b2bf43346f.png)

![captura de tela de 2017-10-09 19-21-33](https://user-images.githubusercontent.com/3011423/31360978-54ee7fca-ad27-11e7-8563-d9222dfcde99.png)

![captura de tela de 2017-10-09 19-21-38](https://user-images.githubusercontent.com/3011423/31360982-54f636de-ad27-11e7-8b7b-246009208340.png)

![captura de tela de 2017-10-09 19-21-50](https://user-images.githubusercontent.com/3011423/31360976-54b1d35e-ad27-11e7-8acf-651504992930.png)

![captura de tela de 2017-10-09 19-21-41](https://user-images.githubusercontent.com/3011423/31360977-54b507ea-ad27-11e7-86b1-bba169f5f4a2.png)

![captura de tela de 2017-10-09 19-21-54](https://user-images.githubusercontent.com/3011423/31360975-54b177ba-ad27-11e7-87cb-aebae8710cf3.png)

![captura de tela de 2017-10-09 19-22-01](https://user-images.githubusercontent.com/3011423/31360974-54b04c82-ad27-11e7-9cc0-12fc18d33df9.png)

![captura de tela de 2017-10-09 19-22-03](https://user-images.githubusercontent.com/3011423/31360973-54ae69c6-ad27-11e7-8283-917cbb1e14b1.png)

**Comment by lonnieezell** [comment]
@natanfelles The `APPPATH` is something I added last night that simplifies values a little bit by replacing the actual values of APPPATH, BASEPATH or ROOTPATH with the word. Makes the URLs a a little shorter for when dealing with smaller screens/tablet views.

The comments idea might work...

**Comment by chistel** [comment]
@natanfelles not every one use bootstrap, i noticed same issue, could not figure out what was wrong, so i commented it. but am thinking if there could be a turn on and off for it, if its thay necessary.

**Comment by puschie286** [comment]
Here is my full login page code
```html
<!DOCTYPE html>
<html>
    <head>
	<title><?= $title ?></title>
	<meta charset="UTF-8">
	<link href="<?= base_url(); ?>css/custom.css" type="text/css" rel="stylesheet" />
	<link href="<?= base_url(); ?>css/materialize.css" type="text/css" rel="stylesheet" />
	<link href="<?= base_url(); ?>css/fonts.css" type="text/css" rel="stylesheet" />
	<style {csp-style-nonce}>
	    body {
			display: table;
			position: absolute;
			height: 100%;
			width: 100%;
	    }
	    .center-container {
			display: table-cell;
			vertical-align: middle;
	    }
	    .center-element {
			margin-left: auto;
			margin-right: auto;
			width: 300px;
	    }
	    .fixed-footer {
			position: fixed;
			bottom: 0;
			left: 0;
			right: 0;
	    }   
	</style>
    </head>
    <body>
	<div class="center-container">
<div class="center-element card-panel" style="min-height: 300px;">
    <h1 style="text-align: center; font-size: 36px; margin-top: 0px;">Anmeldung</h1>
    <?= form_open( base_admin_url() ) ?>
	<div class="input-field">
	    <input type="text" name="username" size="45" <?php if( isset( $username ) ) echo 'value="'.$username.'"'; ?> required />
	    <label>Username</label>
	</div>
	<div class="input-field">
	    <input type="password" name="password" size="72" <?php if( isset( $password ) ) echo 'value="'.$password.'"'; ?> required />
	    <label>Password</label>
	</div>
	<input class="btn" type="submit" value="Login" />
    </form>
</div>
	</div>
	<footer class="fixed-footer page-footer">
	    <div class="container" style="text-align: center;">
		© 2017 Copyright .., Alle Rechte vorbehalten.
	    </div>
	</footer>
    </body>
</html>
```
@natanfelles well yes feels strange, there is nothing special in this code, thats why i created this issue^^
the js thing was just my first idear - havent enough experience with html/css to know all posibilities.
i even dont know if it is posibile to add something that does not affect sibling/child/parent selectors

**Comment by puschie286** [comment]
a i guess i found the problem :
if the first view contain the <!DOCTYPE html> tag and we suround it by a div -> the contents of header and some divs moves out.
dont know why^^but it seems that this is causing the layout destruction

**Comment by puschie286** [comment]
could fixed both problems -> should i push it to #767 or should i create a new pull request ?
